### PR TITLE
Feat/pcc ml subscribe modal ui router

### PR DESCRIPTION
**Pull request by Jisay**
See in app.js diff my proposal to display modal from childs states without being redirected to parent state.
You will see that I define a resolve function with possible modal configurations. Should I set these configurations directly into an attribute of the state declaration?

### CODE REVIEW

File: client/app/dedicatedCloud/mailing-list/subscribe/dedicatedCloud-mailing-list-subscribe.controller.js [hunk 1]
```diff
@@ -1,21 +1,69 @@
...
+
+    onWidzardFinish () {
+        this.loading.subscribe = true;
+
+        return this.dedicatedCloudMailingList.postMailingList(this.model.email, "pcc@ml.ovh.net").then(() =>
+            this.Alerter.success(this.translator.tr("dedicatedCloud_subscribe_mailing_step2_success", this.model.email), "dedicatedCloud")
+        ).catch((error) =>
+            this.Alerter.error([this.translator.tr("dedicatedCloud_subscribe_mailing_step2_error", this.model.email), _.get(error, "message")].join(". "), "dedicatedCloud_alert")
+        ).finally(() => {
+            this.$uibModalInstance.close();
```
Comment (jleveugle) [inline client/app/dedicatedCloud/mailing-list/subscribe/dedicatedCloud-mailing-list-subscribe.controller.js]: You should use UI-Router to change the state ... and the modal will be closed :)

Modal or not should be managed by only one parameter -> `layout: 'modal'`

File: client/app/dedicatedCloud/mailing-list/subscribe/dedicatedCloud-mailing-list-subscribe.controller.js [hunk 2]
```diff
@@ -1,21 +1,69 @@
...
+    /* =====================================
+    =            INITIALIZATION            =
+    ====================================== */
+
+    $onInit () {
+        this.loading.init = true;
+
+        // need to check if user can subscribe to ML
+        return this.dedicatedCloudMailingList.getAvailableMailingLists().then((mailingLists) => {
+            this.canSubscribe = _.indexOf(mailingLists, "pcc@ml.ovh.net") > -1;
```
Comment (jleveugle) [inline client/app/dedicatedCloud/mailing-list/subscribe/dedicatedCloud-mailing-list-subscribe.controller.js]: You cna use `_.some` too :)

File: client/app/dedicatedCloud/mailing-list/subscribe/dedicatedCloud-mailing-list-subscribe.html [hunk 3]
```diff
@@ -1,36 +1,88 @@
...
+<div class="dedicatedCloud-mailing-list-subscribe">
+
+    <div class="oui-modal oui-modal_shadow"
+         data-ng-if="$ctrl.loading.init || !$ctrl.canSubscribe">
+
+        <div class="oui-modal__header">
+            <button type="button"
+                    class="oui-icon oui-icon-close_thin oui-modal__close-button"
+                    aria-label="{{ i18n.wizard_close }}"
+                    data-ng-click="$ctrl.$uibModalInstance.close()">
```
Comment (jleveugle) [inline client/app/dedicatedCloud/mailing-list/subscribe/dedicatedCloud-mailing-list-subscribe.html]: Same comment here, you should change state ;D

File: client/app/dedicatedCloud/mailing-list/subscribe/dedicatedCloud-mailing-list-subscribe.controller.js (hunk 2 above)
Comment (Jisay) [reply client/app/dedicatedCloud/mailing-list/subscribe/dedicatedCloud-mailing-list-subscribe.controller.js]: Indead :-)

Comment (antleblanc) [reply client/app/dedicatedCloud/mailing-list/subscribe/dedicatedCloud-mailing-list-subscribe.controller.js]: Prefer using constant for `pcc@ml.ovh.net'`?

File: package-lock.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (antleblanc) [inline package-lock.json]: by using `yarn` commands, this file shouldn't be generated I guess.

**Pull request closed without merging**